tests/serverpod_test_server/test_integration/server_restart_test.dart

```diff
@@ -0,0 +1,57 @@
+import 'dart:io';
+
+import 'package:serverpod/serverpod.dart';
+import 'package:serverpod_test_server/test_util/mock_stdout.dart';
+import 'package:serverpod_test_server/test_util/test_serverpod.dart';
+import 'package:test/test.dart';
+
+void main() {
+  late Serverpod serverpod;
+
+  setUp(() {
+    serverpod = IntegrationTestServer.create();
+  });
+
+  tearDown(() async {
+    await serverpod.shutdown(exitProcess: false);
+  });
+
+  test(
+      'Given running Serverpod server when it is shutdown and started then it can be successfully started again.',
+      () async {
+    await serverpod.start();
+
+    await serverpod.shutdown(exitProcess: false);
+    var start = serverpod.start();
+
+    await expectLater(start, completes);
```

Comment:
nit: could consider inlining for brevity, just like you do for `session.db.testConnection()` below
```suggestion
    await expectLater(serverpod.start(), completes);
```